pkg/cache/task.go

```diff
@@ -533,6 +533,25 @@ func (task *Task) releaseAllocation() {
 // this reduces the scheduling overhead by blocking such
 // request away from the core scheduler.
 func (task *Task) sanityCheckBeforeScheduling() error {
+	// After version 1.7.0, we should reject the task whose pod is unbound and has conflicting metadata.
+	if !utils.PodAlreadyBound(task.pod) {
+		if err := utils.CheckAppIdInPod(task.pod); err != nil {
+			log.Log(log.ShimCacheTask).Warn("The task has conflicting metadata will be rejected after version 1.7.0.",
```

Comment:
The message should probably be a little different:

"Pod has inconsistent application metadata and may be rejected in a future YuniKorn release"